### ISSUE TITLE
Custom Indexing function for resourceIndexes

### ISSUE THREAD

**Issue by GLosch** [bvaughn/redux-search#78]
Hi, thanks for writing this library. I'm trying to create a custom indexing function to pass into `resourceIndexes`. It doesn't appear that there's documentation on how to accomplish that. I'd be happy to contribute to the documentation if you could please shed some light on how it's meant to be implemented. The only thing I found in the codebase is [here](https://github.com/bvaughn/redux-search/blob/master/source/reduxSearch.js#L15), and StackOverflow hasn't provided any answers.

For some context, I have data in my store that looks something like this:
```
addresses: {
  id: '12345',
  attributes: {
    name: 'some name',
    building_id: '56789'
  }
}
```

I'm looking for a way to index `id` (top-level key), as well as `name` and `building_id` (keys nested one level, under `attributes`). 

Thanks in advance for any help you can offer!

**Comment by bvaughn** [comment]
Unfortunately this library is essentially unmaintained. I haven't used it in years.

If you'd like to dig in and contribute docs, I'd be happy to accept the PRs– but I don't have time to offer any kind of support. (I don't really remember how the library works, and I don't have time to dig in at the moment.)

Sorry. 😦

**Comment by GLosch** [comment]
Thanks for the quick response. If I can get it working, I'll reach out with a documentation PR.

**Comment by GLosch** [comment]
Got it working for my project. There was some documentation around this, after all. For anyone who comes across this same situation, the approach is [here](https://github.com/bvaughn/redux-search/blob/master/docs/reduxSearch.md). 

tl;dr: iterate through the collection with `forEach`, and call `indexDocument` on the key to be indexed.

**Comment by bvaughn** [comment]
Cool 👍 😄
